unblob/models.py

```diff
@@ -155,3 +148,58 @@ def parse_header(self, file: io.BufferedIOBase, endian=Endian.LITTLE):
         header = self._struct_parser.parse(self.HEADER_STRUCT, file, endian)
         logger.debug("Header parsed", header=header)
         return header
+
+
+@attr.define(kw_only=True)
+class Report:
+    """A common base class for different reports"""
+
+    # Stored in `str` rather than `Handler`, because the pickle picks ups structs from `C_DEFINITIONS`
+    handler: Optional[str] = None
+
+    def items(self) -> dict:
+        return attr.asdict(self)
```

Comment:
No, I don't think so.

We want to log the Report's content in the caller side, so I did this:
```python
logger.error("Extract command failed", **error_report.items())
```
The caller does not need to know that the `Report` is an `attrs` class and it should not include `attrs` just for the sake of call `attrs.asdict`. I just called this `items` because of a `dict`-like representation looks easy enough to understand from the caller side.

Reply:
I also like the idea of not leaking the implementation detail that this class is made using attrs.

Reply:
Ok, bought it, but the method name should be `asdict` then, because if you want to call items on the return value, it would look like this: `report.items().items()` which would be very confusing.